src/Microsoft.AspNetCore.Mvc.Core/ModelBinding/ByteArrayModelBinder.cs

```diff
@@ -12,12 +14,13 @@ namespace Microsoft.AspNetCore.Mvc.ModelBinding
     public class ByteArrayModelBinder : IModelBinder
     {
         /// <inheritdoc />
-        public Task<ModelBindingResult> BindModelAsync(ModelBindingContext bindingContext)
+        public Task BindModelAsync(IModelBindingContext bindingContext)
         {
             if (bindingContext == null)
             {
                 throw new ArgumentNullException(nameof(bindingContext));
             }
+            Debug.Assert(bindingContext.Result == null);
```

Comment:
This assert is about public state on a public class that's a parameter. In practice it will never get tripped, but in principle this is either supported (shouldn't assert) or not supported (should throw). I don't feel like these asserts add much value, and one could make the argument that they aren't correct.